### PR TITLE
Qiskit Upgrade and Deprecation Warnings

### LINKED ISSUE #1835: Make mitiq compatible with latest version of Qiskit
Currently Mitiq is built using [Qiskit 0.41.1](https://github.com/unitaryfund/mitiq/blob/b5f3a06069a5b72a19b8b799a4f17803b7ed5589/dev_requirements.txt#L2), and [Qiskit 0.43.0](https://github.com/Qiskit/qiskit-metapackage/releases/tag/0.43.0) has been released. Dependabot [opened a PR](https://github.com/unitaryfund/mitiq/pull/1822) to upgrade our support, but the tests are failing [[example](https://github.com/unitaryfund/mitiq/actions/runs/4958942450/jobs/8872435802)].

Resolution of this issue will amount to upgrading the qiskit version, and making the necessary API changes to qiskit functions so that code coverage remains the same, and the tests pass.

We've also ran into some deprecation warnings in the past in https://github.com/unitaryfund/mitiq/issues/1533. Resolving those would be an added bonus is they are still deprecation warnings in Qiskit 0.43.0.

### PR DESCRIPTION
**Pull request by andre-a-alves**
## Description

Closes #1835 and #1533

Imports `qiskit_aer` directly instead of importing and calling `qiskit.Aer`, and calls `AerSimulator` directly instead of calling `Aer.get_backend()`. These changes allow upgrading Qiskit to 0.43.0 and remove deprecation warnings.

---

### License

- [X] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

Before opening the PR, please ensure you have completed the following where appropriate.
- [ ] I added unit tests for new code.
- [ ] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [ ] I used [Google-style](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) docstrings for functions.
- [ ] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.
- [ ] Added myself / the copyright holder to the AUTHORS file

### CODE REVIEW

**Review by andreamari — APPROVED** [review]
Thanks @andre-a-alves, that was fast and effective! LGTM
@natestemen, I guess this is the first bounty of UnitaryHack.

File: dev_requirements.txt
Comment (natestemen) [inline dev_requirements.txt:1]: Do we need to add `qiskit_aer` as a dependency here?

Comment (andre-a-alves) [reply dev_requirements.txt:1]: At some point, but not yet.

Right now, Qiskit Aer is included as part of the Qiskit metapackage. There are plans to remove it in the future, but as long as Aer is still part of the metapackage, I think it makes sense to just keep them together.

Comment (natestemen) [reply dev_requirements.txt:1]: Oh huh I don't think I knew you could have a package `X` that you pip install, and end with a situation where you can then `import X`, _and_ `import Y`. TIL.

**Comment by andre-a-alves** [comment]
Yesterday while working on the docs, I realized they still used the old interface. I added a commit that corrects that.
I wasn't sure whether `IBMProvider` needed to be included int he dev requirements, since it's only necessary for the code in the docs, so I went ahead and added it, but I can easily remove it as well.

File: docs/source/examples/cirq-ibmq-backends.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ from mitiq import zne
 from mitiq.interface.mitiq_qiskit.qiskit_utils import initialized_depolarizing_noise
 
 
-USE_REAL_HARDWARE = False
+USE_REAL_HARDWARE = True
```
Comment (natestemen) [inline docs/source/examples/cirq-ibmq-backends.md:37]: I don't think we want the docs to try to use real hardware, do we?

Comment (natestemen) [reply docs/source/examples/cirq-ibmq-backends.md:37]: Actually this is causing a failure on RTD: https://mitiq--1847.org.readthedocs.build/en/1847/examples/cirq-ibmq-backends.html

**Review by natestemen — CHANGES_REQUESTED** [review]

File: docs/source/examples/cirq-ibmq-backends.md (hunk 1 above)
Comment (andre-a-alves) [reply docs/source/examples/cirq-ibmq-backends.md]: Oh! Sorry about that. I had switched them to real hardware to test some of the changes and make sure IBMProvider worked. Latest merge fixes this.

**Comment by Misty-W** [comment]
@natestemen, once @andre-a-alves resolves the conflicts, is this good to go?

**Review by natestemen — APPROVED** [review]
🚢 Thanks Andre!

**Pull request merged**